lib/Logger.js

```diff
@@ -43,6 +45,26 @@ module.exports = class Logger {
     this._output({ level, record });
   }
 
+  drain(timeout) {
+    this._draining =
+      this._draining ||
+      new Promise((resolve, reject) => {
```

Comment:
It's kind of funky assigning a promise to this._draining since the name implies a boolean. I don't want to manage two properties, but maybe a better name is called for

Reply:
Leaving as is